src/main/java/org/stellar/sdk/TimeBounds.java

```diff
@@ -11,24 +11,44 @@
 final public class TimeBounds {
 	final private long mMinTime;
 	final private long mMaxTime;
-	
+
 	/**
 	 * @param minTime 64bit Unix timestamp
 	 * @param maxTime 64bit Unix timestamp
 	 */
 	public TimeBounds(long minTime, long maxTime) {
-		if (maxTime > 0 && minTime >= maxTime) {
+		if (minTime < 0) {
+			throw new IllegalArgumentException("minTime cannot be negative");
+		}
+
+		if (maxTime < 0) {
+			throw new IllegalArgumentException("maxTime cannot be negative");
+		}
+
+		if (maxTime != 0 && minTime > maxTime) {
 			throw new IllegalArgumentException("minTime must be >= maxTime");
 		}
-		
+
 		mMinTime = minTime;
 		mMaxTime = maxTime;
 	}
-	
+
+	/**
+	 * A factory method that sets maxTime to the specified second from now.
+	 *
+	 * @param timeout Timeout in seconds.
+	 * @return TimeBounds
+	 */
+	public static TimeBounds afterNow(long timeout) {
```

Comment:
[Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) was added in Java 8, so we cannot use it for compatibility.